.github/workflows/docs.yaml

```diff
@@ -0,0 +1,40 @@
+name: Quarto documents
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  install-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: quarto-dev/quarto-actions/setup@v2
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: cache poetry
+        uses: actions/cache@v4
+        with:
+          path: ~/.local
+          key: ${{ runner.os }}-poetry
+
+      - name: install poetry
+        run: pip install poetry
+
+      - name: install package
+        run: poetry install -C model
+
+      - name: installing other dependencies
+        run: poetry run -C model pip install pyyaml nbclient nbformat
+
+      - name: Render documents
+        run: |
+            cd model
+            for file in $(find docs -name '*.qmd'); do
+                poetry run quarto render $file
+            done
```

Comment:
I don't think this will actually commit the newly rendered docs to the repo, but maybe I've misunderstood the goal here.

Reply:
Yes, this is not commiting the rendered docs, so it only checks that they render. In the future, I hope to use this as a reference for us to create the rst documents. Perhaps we can do that now?

Reply:
I have no issue with just checking that the docs rendering successfully if that was the goal.

My two thoughts:

1. If we're still intending to have github-flavored Markdown versions of the rendered tutorials tracked in the repo, why not commit the versions rendered in CI?
2.  That said, there's an argument that we shouldn't have rendered tutorials in the repo, namely the Python adage that ["There should be one-- and preferably only one --obvious way to do it."](https://peps.python.org/pep-0020/). With that in mind, perhaps the rendered tutorials should exist _only_ on the documentation website.

Reply:
1. I was thinking about that. The one problem I see is when you are working on that file, you'd like to compile it to ensure it is running properly.
2. I see the point, perhaps have `model/docs/README.md` pointing to the website with the compiled docs and remove the `md` files from the tree (just keep qmd)?

I'll open an issue for this.